autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +385,220 @@ model UserBlockCredit {
+
+  hasInitiatedChargeBackProcess Boolean          @default(false)
+  chargeBackProcessInitiatedAt  DateTime?
+  UserAccounting                UserAccounting[]
+
+  @@index([isSubscribed, isCancelling])
+}
+
+// This table tracks the different subscription plans available to a user
+model SubsctiptionPlan {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name        String
+  description String
+  planIconUrl String?
+
+  creditsPerMonth Int
+  usdPrice        Float
+
+  isDeleted        Boolean            @default(false)
+  UserSubscription UserSubscription[]
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+//////////////   Store TABLES    /////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model Profile {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isGroupProfile Boolean    @default(false)
+  groupId        String?
+  group          UserGroup? @relation(fields: [groupId], references: [id])
+
+  username    String @unique
+  description String
+
+  website  String?
+  github   String?
+  linkedin String?
+  facebook String?
+  twitter  String?
+
+  avatarUrl String?
+
+  @@index([username])
+}
+
+model StoreListing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  isDeleted  Boolean @default(false)
+  // Not needed but makes lookups faster
+  isApproved Boolean @default(false)
+
+  slug        String
+  name        String
+  description String
+
+  agentId      String
+  agentVersion Int
+  agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version])
+
+  owningUserId String
+  user         User   @relation(fields: [owningUserId], references: [id], onDelete: Cascade)
+
+  isGroupListing Boolean    @default(false)
+  owningGroupId  String?
+  owningGroup    UserGroup? @relation(fields: [owningGroupId], references: [id])
+
+  StoreListingVersions   StoreListingVersion[]
+  StorePricings          StorePricing[]
+  UserBlockCredit        UserBlockCredit[]
+  StoreListingSubmission StoreListingSubmission[]
+
+  @@index([isApproved])
+  @@index([agentId])
+  @@index([owningUserId])
+  @@index([owningGroupId])
+}
+
+model StorePricing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  // Owner can be found from the storeListing
+  storeListingId String
+  storeListing   StoreListing @relation(fields: [storeListingId], references: [id])
+
+  canPurchase   Boolean @default(true)
+  pruchasePrice Float?
+
+  commissionPerRun Float?
```

Comment:
This is a bit at odds with our block transaction accounting. We need to decide how this commission is applied. Is it only applied on a successful run?

Reply:
Let's chat about this